src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -740,7 +764,7 @@ public void SetNickNameFromAttribute()
         public event Action<NodeModel> Modified;
         public virtual void OnNodeModified(bool forceExecute = false)
         {
-            if (!RaisesModificationEvents)
+            if (!RaisesModificationEvents || IsFrozen)
```

Comment:
We have NodeExecutionTest class. If a node is frozen and modified, they will not execute and the result remains the same.All the tests  in NodeExeuctionTest class, tests this functionality.